### PR TITLE
Add custom 404 page

### PR DESCRIPTION
**Pull request by nkprince007**
**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [x] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [x] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `cobot mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

**Comment by nkprince007** [comment]
screen shot 2017-08-31 at 12 15 50 am
screen shot 2017-08-31 at 12 49 36 am
screen shot 2017-08-31 at 12 49 41 am
screen shot 2017-08-31 at 12 49 45 am
screen shot 2017-08-31 at 12 49 48 am
screen shot 2017-08-31 at 12 49 53 am
screen shot 2017-08-31 at 12 49 58 am
screen shot 2017-08-31 at 12 50 06 am
screen shot 2017-08-31 at 12 50 12 am

File: 404.html [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  <style>
+    .hidden-until-ready {
+      visibility: hidden;
+    }
+  </style>
+  <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.1.0/jquery.min.js"></script>
+  <script src="https://cdnjs.cloudflare.com/ajax/libs/materialize/0.97.8/js/materialize.min.js"></script>
+  <link rel="stylesheet" href="resources/vendors/materialize/css/materialize.css">
+  <link rel="stylesheet" href="resources/css/404.css">
+  <link rel="stylesheet" href="https://rawgit.com/coala/coalaCSS/master/coala.css">
```
Comment (hemangsk) [inline 404.html]: +1 for this

Comment (hemangsk) [reply 404.html]: ;)

File: 404.html [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  <link href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" rel="stylesheet">
+  <link rel="shortcut icon" href="favicon.ico" type="image/x-icon">
+</head>
+<body class="hidden-until-ready">
+  <main>
+    <div class="coala-book container">
+      <img src="resources/images/coala-404.png" alt="missing coala in shock">
+    </div>
+    <div class="error container">
+      <h3 class="message">Oops! That was unexpected...</h3>
```
Comment (hemangsk) [inline 404.html]: Suggestion : try adding class "light"

File: resources/css/404.css [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+}
+.error p span {
+    font-weight: 400;
+}
+.error .home {
+    text-transform: uppercase;
+    border-radius: 5px;
+}
+.error .buttons {
+    padding: 3%;
```
Comment (hemangsk) [inline resources/css/404.css]: Border radius 0 for crawl back to home button. That'll keep consistency.

**Comment by hemangsk** [comment]
ack acafdc4

**Pull request merged**